providers/module.rb

```diff
@@ -9,18 +9,21 @@ def whyrun_supported?
 
 # Compile and deploy module (also upgrade)
 action :deploy do
-  filename="semodule-#{new_resource.name}"
-  f = file "#{Chef::Config[:file_cache_path]}/#{filename}.te" do
+  directory "#{Chef::Config[:file_cache_path]}/semodule-#{new_resource.name}" do
+    action :create
+  end
```

Comment:
I'm inferring that you're populating the directory using some other resource.
If so, why create the directory? If anything, not creating the directory will cause the module compilation to fail, which is a great indicator that something's faulty (rather than creating an empty module?)

Also, this needs documentation. Could you update the README?